docker/docker_image.go

```diff
@@ -19,7 +19,17 @@ type Image struct {
 // NewDockerImage returns a new Image interface type after setting up
 // a client to the registry hosting the given image.
 func NewDockerImage(img, certPath string, tlsVerify bool) (types.Image, error) {
-	s, err := newDockerImageSource(img, certPath, tlsVerify)
+	ref, err := parseReference("//" + img)
+	if err != nil {
+		return nil, err
+	}
+	return newDockerImage(ref, certPath, tlsVerify)
+}
```

Comment:
I was feeling the same way (when working on client abstraction)

> and any caller knowledgeable enough about a transport to directly call one of these functions might be knowledgeable enough to give it a fully parsed specification (reference.Named + tag for Docker, namespace, stream, tag for OpenShift) etc.

+1

> Arguably anyone starting with a string should go through https://github.com/projectatomic/skopeo/pull/138/files#diff-6179b09a11bd9337d53f297f715ac27dR13

and +1